tests/reporter/kubernetes.go

```diff
@@ -516,6 +525,63 @@ func (r *KubernetesReporter) logEvents(virtCli kubecli.KubevirtClient, since tim
 	fmt.Fprintln(f, string(j))
 }
 
+func (r *KubernetesReporter) logSRIOVInfo(virtCli kubecli.KubevirtClient) {
+	sriovOutputDir := filepath.Join(r.artifactsDir, "sriov")
+	if err := os.MkdirAll(sriovOutputDir, 0777); err != nil {
+		fmt.Fprintf(os.Stderr, "failed to create directory: %v\n", err)
+		return
+	}
+
+	r.logSRIOVNodeState(virtCli, sriovOutputDir)
+	r.logSRIOVNodeNetworkPolicies(virtCli, sriovOutputDir)
+	r.logSRIOVNetworks(virtCli, sriovOutputDir)
+	r.logSRIOVOperatorConfigs(virtCli, sriovOutputDir)
+}
+
+func (r *KubernetesReporter) logSRIOVNodeState(virtCli kubecli.KubevirtClient, outputFolder string) {
+	nodeStateLogPath := filepath.Join(outputFolder, fmt.Sprintf("%d_nodestate.log", r.failureCount))
+	r.dumpK8sEntityToFile(virtCli, sriovNodeStateEntity, v1.NamespaceAll, nodeStateLogPath)
+}
+
+func (r *KubernetesReporter) logSRIOVNodeNetworkPolicies(virtCli kubecli.KubevirtClient, outputFolder string) {
+	nodeNetworkPolicyLogPath := filepath.Join(outputFolder, fmt.Sprintf("%d_nodenetworkpolicies.log", r.failureCount))
+	r.dumpK8sEntityToFile(virtCli, sriovNodeNetworkPolicyEntity, v1.NamespaceAll, nodeNetworkPolicyLogPath)
+}
+
+func (r *KubernetesReporter) logSRIOVNetworks(virtCli kubecli.KubevirtClient, outputFolder string) {
+	networksPath := filepath.Join(outputFolder, fmt.Sprintf("%d_networks.log", r.failureCount))
+	r.dumpK8sEntityToFile(virtCli, sriovNetworks, v1.NamespaceAll, networksPath)
+}
+
+func (r *KubernetesReporter) logSRIOVOperatorConfigs(virtCli kubecli.KubevirtClient, outputFolder string) {
+	operatorConfigPath := filepath.Join(outputFolder, fmt.Sprintf("%d_operatorconfigs.log", r.failureCount))
+	r.dumpK8sEntityToFile(virtCli, sriovOperatorConfigs, v1.NamespaceAll, operatorConfigPath)
+}
+
+func (r *KubernetesReporter) dumpK8sEntityToFile(virtCli kubecli.KubevirtClient, entityName string, namespace string, outputFilePath string) {
+	requestURI := fmt.Sprintf(sriovEntityURITemplate, namespace, entityName)
+	f, err := os.OpenFile(outputFilePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to open file: %v\n", err)
+		return
+	}
+	defer f.Close()
+
+	response, err := virtCli.RestClient().Get().RequestURI(requestURI).Do().Raw()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to dump SR-IOV [%s] entities: %v\n", entityName, err)
```

Comment:
The method is not SR-IOV specific. If somebody uses it for a different objects, this log would be misleading

Reply:
Done.